src/components/menuFilter/menu-filter.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * @ngdoc module
+ * @name material.components.menuFilter
+ */
+
+angular.module('material.components.menuFilter', [
+  'material.core',
+])
+  .directive('mdMenuFilter', MdMenuFilterDirective);
+
+/**
+ * @ngdoc directive
+ * @name mdMenuFilter
+ * @module material.components.menuFilter
+ * @restrict E
+ *
+ * @description
+ * The menu filter directive is used to make adding a searchbar to
+ * the top of a component quick and painless.
+ *
+ * @param {string=} aria-label Optional label for accessibility. Only necessary if no placeholder or
+ *   explicit label is present.
+ * @param {Array=} filtered-items The variable the filtered
+ *   results will be stored in.
+ * @param {Array=} items The items to filter.
```

Comment:
Maybe using input vs output would help explain the difference between filteredItems and items. You should also be explict about which you are watching for updates. For instance, if I change filteredItems, what will happen in the UI?